include/thingset/isotp_fast.h

```diff
@@ -130,14 +204,34 @@ int isotp_fast_recv(struct isotp_fast_ctx *ctx, struct can_filter sender, uint8_
  * @param ctx The bound context on which the message should be sent
  * @param data A pointer to the data containing the message to send
  * @param len The length of the data in @ref data
- * @param target_addr The node ID identifying the recipient. This will be
- *                    combined with the sending address @ref my_addr on @ref
- *                    ctx to form the CAN ID on the message.
- * @param target_bus Target bus number (4-bit value) to send the message to.
+ * @param target_addr The CAN ID identifying the recipient.
  * @param sent_cb_arg A pointer to data to be supplied to the callback
  *                    that will be invoked when the message is sent.
  *
  * @returns 0 on success.
  */
 int isotp_fast_send(struct isotp_fast_ctx *ctx, const uint8_t *data, size_t len,
-                    const uint8_t target_addr, uint8_t target_bus, void *sent_cb_arg);
+                    const struct isotp_fast_addr target_addr, void *sent_cb_arg);
+
+#ifdef CONFIG_ISOTP_FAST_FIXED_ADDRESSING
+struct isotp_fast_addr isotp_fast_get_counterpart_addr(const struct isotp_fast_addr addr);
```

Comment:
Pointers are more efficient.

```suggestion
struct isotp_fast_addr isotp_fast_get_counterpart_addr(const struct isotp_fast_addr *addr);
```